src/Pirouette/Term/Symbolic/Eval.hs

```diff
@@ -166,7 +166,7 @@ path x st =
 -- | A 'SymEvalT' is equivalent to a function with type:
 --
 -- > SymEvalSt lang -> SMT.Solver -> m [(a, SymEvalSt lang)]
-newtype SymEvalT lang m a = SymEvalT {symEvalT :: StateT (SymEvalSt lang) (SMT.SolverT (ListT m)) a}
+newtype SymEvalT lang m a = SymEvalT {symEvalT :: StateT (SymEvalSt lang) (SMT.SolverT (WeightedT Integer m)) a}
```

Comment:
If this does the trick, it will be what we would call a pro-gamer move! :crossed_fingers: :astonished:

Reply:
Everything still works, but there's no usage of the weights yet, so the results are the same (and the performance suffers a bit because before we were using lists).